runtime/src/bank/check_transactions.rs

```diff
@@ -102,36 +116,51 @@ impl Bank {
                 nonce: None,
                 lamports_per_signature: hash_info.lamports_per_signature(),
             })
-        } else if let Some((nonce, nonce_data)) =
-            self.check_and_load_message_nonce_account(tx.message(), next_durable_nonce)
-        {
+        } else if let Some(nonce) = self.check_load_and_advance_message_nonce_account(
+            tx.message(),
+            next_durable_nonce,
+            lamports_per_signature,
+        ) {
             Ok(CheckedTransactionDetails {
                 nonce: Some(nonce),
-                lamports_per_signature: nonce_data.get_lamports_per_signature(),
+                lamports_per_signature,
```

Comment:
This is technically wrong but currently harmless (because this lps value is only used for turning off fees inside `validate_transaction_fee_payer`). Let's play it safe and keep the old logic of using the previous nonce data lps value rather than changing it to the blockhash queue value.

Reply:
ok i believe we should be perfectly in line with master now. the nonce data is updated with the lps value from the tip of the blockhash queue. but the previous nonce data lps value is assigned in the check details, which is ultimately only compared to 0 in svm to see if fees are in play

Reply:
Cool that matches my understanding as well, thanks for making the change